app/grandchallenge/components/tasks.py

```diff
@@ -227,6 +227,71 @@ def remove_container_image_from_registry(
     model = apps.get_model(app_label=app_label, model_name=model_name)
     instance = model.objects.get(pk=pk)
 
+    from grandchallenge.algorithms.models import AlgorithmImage, Job
+    from grandchallenge.evaluation.models import Evaluation, Method
+    from grandchallenge.workstations.models import Session, WorkstationImage
+
+    instance_in_use = False
+
+    if isinstance(instance, Method):
+        instance_in_use = (
+            Evaluation.objects.filter(
+                method=instance,
+            )
+            .exclude(
+                status__in=[
+                    Evaluation.SUCCESS,
+                    Evaluation.FAILURE,
+                    Evaluation.CANCELLED,
+                ]
+            )
+            .exists()
+        )
+
+    elif isinstance(instance, AlgorithmImage):
+        instance_in_use = (
+            Evaluation.objects.filter(
+                submission__algorithm_image=instance,
+            )
+            .exclude(
+                status__in=[
+                    Evaluation.SUCCESS,
+                    Evaluation.FAILURE,
+                    Evaluation.CANCELLED,
+                ]
+            )
+            .exists()
+            or Job.objects.filter(
+                algorithm_image=instance,
+            )
+            .exclude(
+                status__in=[
+                    Job.SUCCESS,
+                    Job.FAILURE,
+                    Job.CANCELLED,
+                ]
+            )
+            .exists()
+        )
+
+    elif isinstance(instance, WorkstationImage):
+        instance_in_use = (
+            Session.objects.filter(
+                workstation_image=instance,
+            )
+            .exclude(
+                status__in=[
+                    Session.RUNNING,
```

Comment:
I excluded the RUNNING status from filter because I believe at this point, the image is already pulled from the registry and the workstation container is up and running and thus the image is not in use. Am I right?